Extend contribution notes, add submodule information, change syntax and fix typos

Lupphes opened the pull request:
I created this pull request to add contribution notes to the documentation. While writing the changes, I also added the required information about the common repository #12 and fixed some typos and updated the markdown syntax.

Lupphes commented:
Relates #12

phracek commented on README.md:
```diff
@@ -1,58 +1,52 @@
+# Welcome to the SCL.org Space
 
-Why SCL.org? Since all the repositories, under this organization are somehow connected to the Software Collections concept. Software Collections give you the power to build, install, and use multiple versions of software on the same system, without affecting system-wide installed packages.
+This is an overview of the sclorg organization, aka all one can find here.
```
Please do not use 'aka'. Use the full names.

Lupphes replied:
I rewrote some sentences to make them sound more professional :smile:

phracek commented on README.md:
```diff
@@ -1,58 +1,52 @@
-Why SCL.org? Since all the repositories, under this organization are somehow connected to the Software Collections concept. Software Collections give you the power to build, install, and use multiple versions of software on the same system, without affecting system-wide installed packages.
+This is an overview of the sclorg organization, aka all one can find here.
 
-And since we have had set of new web development tools available as RPMs, tested and available in the directory on [softwarecollections.org](https://www.softwarecollections.org/en/), it is a good idea to re-use those tested packages and make container images from them, that are working fine in OpenShift.
```
Why did you delete the reference? It is not valid?

Lupphes replied:
It is still there on line 7. Git diff moved it to the wrong line (under removed Repositories Overview).

phracek reviewed (changes requested):
Please address my changes if they are feasible.

pkubatrh commented on contribution.md:
```diff
@@ -81,11 +82,25 @@ In the created repositories add also robot account which is mandatory for publis
+1. Fork the repository
+2. Run `git submodule update --init` to download the `common` submodule with `common/common.mk` makefile. We use this submodule for common tests and functions across all our repositories.
+3. Add a feature or bug fix in the templates (src directory) or values (specs/multispec.yml file)
+   1. If you do not change anything in those directories, you do not need to regenerate files
```
This step is not typical at all - we only use distgen generation in a small number of repositories. I would just go with a generic "Add a feature or bug fix" step. If you want to, you can explicitly state that for distgen-enabled repositories the sources are in the `/src` directory and so on.

Lupphes replied:
Fixed in the last commit. Mentioned with `*`.

Lupphes committed:
docs: Rewrite documentation README.md and contribution.mb
Extend contribution notes and add submodule information
Fix typos in the documentation
Update markdown syntax

zmiklank commented on README.md:
```diff
@@ -1,58 +1,53 @@
+# Welcome to the SCL.org Space
 
-Why SCL.org? Since all the repositories, under this organization are somehow connected to the Software Collections concept. Software Collections give you the power to build, install, and use multiple versions of software on the same system, without affecting system-wide installed packages.
+This document offers a comprehensive introduction to the sclorg organization, highlighting the repositories under our umbrella and providing examples of their practical applications, along with links to relevant external resources.
```
```suggestion
This document offers a comprehensive introduction to the Software Collections Organization (SCLOrg), highlighting the repositories under our umbrella and providing examples of their practical applications, along with links to relevant external resources.
```

zmiklank commented on README.md:
```diff
@@ -1,58 +1,53 @@
 
-Repositories Overview
----------------------
+In light of recent developments, we have introduced a series of RPMs web development tools. These tools have been thoroughly tested and are accessible via the softwarecollections.org directory. To maximize the utility of these verified packages, we encourage the creation of container images compatible with OpenShift.
```
I have no information about publishing our work on softwarecollections.org. Maybe this was in the history and works no longer for us?
@pkubatrh do you know something more about this, please?

zmiklank replied:
Thanks for removing the "softwarecollections.org". However, I think that removing it causes the sentence to loose its meaning. I suggest removing whole sentence.

zmiklank replied:
But wait a little please, I will rather ask Peto about this.

zmiklank replied:
So the result is we really can remove it.
I saw one more occurrence of softwarecollections.org in README.md. Would you mind deleting all the occurrences please?

Lupphes replied:
If it is confirmed, I can remove it. Thank you for the review. I will commit the changes.

zmiklank commented on README.md:
```diff
@@ -1,58 +1,53 @@
-* [RHSCL Container CI](https://github.com/sclorg/rhscl-container-ci): Scripts that help testing containers on RHEL platform.
+## Repository Overview
 
+- [Container Contribution Guidelines](contribution.md): Learn about what principles to follow when contributing to the sclorg organization, particularly in relation to container repositories.
```
```suggestion
- [Container Contribution Guidelines](contribution.md): Learn about what principles to follow when contributing to the SCLOrg, particularly in relation to container repositories.
```
AFAIK, in general we use capital letters like this for SCLOrg. This applies also for other occurrences of 'sclorg' in this text.

zmiklank commented on README.md:
```diff
@@ -1,58 +1,53 @@
- * https://fedoramagazine.org/quick-containers-with-fedora-dockerfiles/
- * https://www.liquidweb.com/kb/how-to-list-and-attach-to-docker-containers/
- * https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux_atomic_host/7/html/overview_of_containers_in_red_hat_systems/introduction_to_linux_containers
+# Where to Learn More About Containers
```
In my opinion is redundant to state such a generic information in SCLOrg's readme.

zmiklank commented on README.md:
```diff
@@ -1,58 +1,53 @@
- * https://docs.docker.com/engine/reference/builder/
- * https://developers.redhat.com/blog/2016/02/24/10-things-to-avoid-in-docker-containers/
- * http://docs.projectatomic.io/container-best-practices/
+## How to write a Dockerfile
```
The same for writing dockerfile - this is imo not information, that should be stated in SCLOrg's readme, as is not really related to SCLOrg stuff.

zmiklank commented on contribution.md:
```diff
@@ -1,117 +1,135 @@
 
-All changes done in the images go through pull request (PR) and require review from another maintainer before merging. A reasonably small change that is extra-obvious and cannot cause any issue can be merged/committed without a review (typically typos in documentation).
-
+All modifications to the repositories and images must be submitted via a pull request (PR) and undergo review by another maintainer before merging. Exceptionally, small and obvious changes that pose no risk (such as fixing typos in documentation) may be merged or committed without review.
```
```suggestion
All modifications to the repositories and images must be submitted via a Pull Request (PR) and undergo review by another maintainer before merging.
```
Pull Request is in general written with first capital letters. Please check also other occurrences in the text and unify them.

zmiklank commented on images.md:
```diff
@@ -73,7 +68,6 @@ The following is a list of container images maintained in this organization. Mos
 | [2.4](https://github.com/sclorg/httpd-container/tree/master/2.4) | [rhscl/httpd-24-rhel7](https://access.redhat.com/containers/#/registry.access.redhat.com/rhscl/httpd-24-rhel7) | [centos/httpd-24-centos7](https://hub.docker.com/r/centos/httpd-24-centos7/) |
 
-
 ## Thermostat
```
We do not really maintain this anymore, could be probably removed.

zmiklank commented on images.md:
```diff
@@ -87,7 +81,6 @@ The following is a list of container images maintained in this organization. Mos
 | [s2i](https://github.com/sclorg/s2i-base-container.git) | [rhscl/s2i-base-rhel7](https://access.redhat.com/containers/#/registry.access.redhat.com/rhscl/s2i-base-rhel7) | [centos/s2i-base-centos7](https://hub.docker.com/r/centos/s2i-base-centos7/) |
 
-
 ## Developer Toolset
```
The same stands for developer toolset.

zmiklank commented on images.md:
```diff
@@ -97,15 +90,13 @@ The following is a list of container images maintained in this organization. Mos
 | [5](https://github.com/sclorg/redis-container/tree/master/5) | [rhscl/redis-5-rhel7](https://access.redhat.com/containers/#/registry.access.redhat.com/rhscl/redis-5-rhel7) |  |
 
-
 ## MongoDB
```
And also for MongoDB.
We support following container repositories:
httpd-container
nginx-container
s2i-php-container
varnish-container
mariadb-container
postgresql-container
s2i-nodejs-container
s2i-python-container
mysql-container
redis-container
s2i-perl-container
s2i-ruby-container

zmiklank commented on contribution.md:
```diff
@@ -1,117 +1,135 @@
 
-The images in this organization include test-suites, that test the images in basic use cases.
-The tests are being run on [Testing Farm](https://docs.testing-farm.io/general/0.1/index.html). Testing is done before the change is merged into the master branch. Therefore the place for triggering tests on demand is in a Pull Request.
+This organisation's images include test suites, which test the images in basic use cases.
```
I see occurrences of 'organisation' and 'organization' in this text. This should be united.

zmiklank commented on README.md:
```diff
@@ -1,58 +1,53 @@
-Welcome to the SCL.org Space
-============================
-This is basically an overview around the sclorg organization, aka what all one can find here.
+# Welcome to the SCL.org Space
```
```suggestion
# Welcome to the Software Collections Organization
```

zmiklank commented on README.md:
```diff
@@ -1,58 +1,53 @@
+This document offers a comprehensive introduction to the sclorg organization, highlighting the repositories under our umbrella and providing examples of their practical applications, along with links to relevant external resources.
 
-And since we have had set of new web development tools available as RPMs, tested and available in the directory on [softwarecollections.org](https://www.softwarecollections.org/en/), it is a good idea to re-use those tested packages and make container images from them, that are working fine in OpenShift.
+The primary focus of SCL.org revolves around the Software Collections concept. Software Collections offer a streamlined approach to building, installing, and utilizing multiple software versions on a single system, without disrupting pre-existing system-wide package installation.
```
```suggestion
The primary focus of SCLOrg revolves around the Software Collections concept. Software Collections offer a streamlined approach to building, installing, and utilizing multiple software versions on a single system, without disrupting pre-existing system-wide package installation.
```

zmiklank reviewed (changes requested):
Thank you for this PR. Please see my notes.

zmiklank commented on contribution.md:
```diff
@@ -1,117 +1,135 @@
-# Contribution to Container Images
-
-This document describes rules and best practices for contributing to http://github.com/sclorg/.
+# Contributing to Container Images
```
This is good opportunity to explain usage of "ready for review" label in contribution.md. Could you please state it somewhere here and also link a query for searching all open PRs in SCLOrg labeled with this label?

zmiklank replied:
https://github.com/search?q=org%3Asclorg+label%3A%22ready+for+review%22+state%3Aopen&type=Issues

This is the query. Could you please double-check if it works as intended?

Lupphes committed:
docs: Fix grammatical mistakes and consistency
Fix naming
Add label usage in the repositories
Mark deprecated images and add new versions

Co-authored-by: zmiklank <zmiklank@redhat.com>

zmiklank reviewed (approved):
Thank you for your work. LGTM.
